x/group/internal/math/dec_test.go

```diff
@@ -0,0 +1,286 @@
+package math
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"pgregory.net/rapid"
+)
+
+func TestDec(t *testing.T) {
+
+	// Property tests
+	t.Run("TestNewDecFromInt64", rapid.MakeCheck(testDecInt64))
+
+	// Properties about addition
+	t.Run("TestAddLeftIdentity", rapid.MakeCheck(testAddLeftIdentity))
+	t.Run("TestAddRightIdentity", rapid.MakeCheck(testAddRightIdentity))
+	t.Run("TestAddCommutative", rapid.MakeCheck(testAddCommutative))
+	t.Run("TestAddAssociative", rapid.MakeCheck(testAddAssociative))
+
+	// Properties about subtraction
+	t.Run("TestSubRightIdentity", rapid.MakeCheck(testSubRightIdentity))
+	t.Run("TestSubZero", rapid.MakeCheck(testSubZero))
+
+	// Properties combining operations
+	t.Run("TestSubAdd", rapid.MakeCheck(testSubAdd))
+	t.Run("TestAddSub", rapid.MakeCheck(testAddSub))
+
+	// Properties about comparision and equality
+	t.Run("TestCmpInverse", rapid.MakeCheck(testCmpInverse))
+	t.Run("TestEqualCommutative", rapid.MakeCheck(testEqualCommutative))
+
+	// Properties about tests on a single Dec
+	t.Run("TestIsNegative", rapid.MakeCheck(testIsNegative))
+
+	// Unit tests
+	zero := Dec{}
+	one := NewDecFromInt64(1)
+	two := NewDecFromInt64(2)
+	three := NewDecFromInt64(3)
+	four := NewDecFromInt64(4)
+	five := NewDecFromInt64(5)
+	minusOne := NewDecFromInt64(-1)
+
+	onePointOneFive, err := NewDecFromString("1.15")
+	require.NoError(t, err)
+	twoPointThreeFour, err := NewDecFromString("2.34")
+	require.NoError(t, err)
+	threePointFourNine, err := NewDecFromString("3.49")
+	require.NoError(t, err)
+	onePointFourNine, err := NewDecFromString("1.49")
+	require.NoError(t, err)
+	minusFivePointZero, err := NewDecFromString("-5.0")
+	require.NoError(t, err)
+
+	res, err := two.Add(zero)
+	require.NoError(t, err)
+	require.True(t, res.IsEqual(two))
+
+	res, err = five.Sub(two)
+	require.NoError(t, err)
+	require.True(t, res.IsEqual(three))
+
+	res, err = onePointOneFive.Add(twoPointThreeFour)
+	require.NoError(t, err)
+	require.True(t, res.IsEqual(threePointFourNine))
+
+	res, err = threePointFourNine.Sub(two)
+	require.NoError(t, err)
+	require.True(t, res.IsEqual(onePointFourNine))
+
+	res, err = minusOne.Sub(four)
+	require.NoError(t, err)
+	require.True(t, res.IsEqual(minusFivePointZero))
+
+	require.False(t, zero.IsNegative())
+
+	require.False(t, one.IsNegative())
+
+	require.True(t, minusOne.IsNegative())
+}
+
+// TODO: Think a bit more about the probability distribution of Dec
```

Comment:
nit: Let's not have TODOs w/o having issue reference(s) OR fixing/removing the TODO in the PR.